### PR TITLE
ci/openshift-ci: Use loop instead of curl --retry-all-errors

### PR DESCRIPTION
**Pull request by ldoktor**
The --retry-all-errors argument is not supported on the CI runners, use a busy loop instead.

### CODE REVIEW

**Comment by ldoktor** [comment]
@wainersm @gkurz the selinux enablement PR used `curl --retry-all-errors` which is not supported on the CI runners, this should work.

File: .ci/openshift-ci/run_smoke_test.sh [hunk 1]
```diff
@@ -56,7 +56,11 @@ else
     port=$(oc get service/http-server-service -o jsonpath='{.spec.ports[0].nodePort}')
 fi
 
-curl "${host}:${port}${hello_file}" -s -o hello_msg.txt --retry 60 --retry-delay 1 --retry-all-errors
+rm -f hello_msg.txt
+for i in $(seq 60); do
```
Comment (wainersm) [inline .ci/openshift-ci/run_smoke_test.sh:60]: hi @ldoktor , good catch.

One suggestion is to use the `waitForProcess` (e.g. `waitForProcess $wait_time $sleep_time "$cmd"`) utility.

Comment (ldoktor) [reply .ci/openshift-ci/run_smoke_test.sh:60]: Sure

**Comment by ldoktor** [comment]
Changes:

* use waitForProcess
* new commit to remove the pre-existing file still consider a hot-fix, I'd like to remove the file completely just wait for the pipeline to stabilize first
* new commit to avoid failures as by default it raises exception on non-zero return unless treated which prevents the cleanup execution (which is something I'd also want to properly address later)

**Review by wainersm — APPROVED** [review]
Thanks @ldoktor !

**Comment by wainersm** [comment]
@beraldoleal mind to review this one?

**Comment by ldoktor** [comment]
Changes: Added the fixes to commit a message

**Comment by ldoktor** [comment]
Changes: Removed the controversial commit that treats `grep` exception as it's not really related to the changes that are required no.

**Review by gkurz — APPROVED** [review]
LGTM. Thanks @ldoktor !

**Pull request merged**